### PR TITLE
LVS hiera support

### LINKED ISSUE #116: Hiera lookups and this module
I spent some time working with this module to get it to setup up my LVS and I got it to setup the configuration files. 

Below is a fragment of my Hiera file. This goes probably to my still incomplete understanding of puppet and how it sees hiera data as to why I am still a bit confused as to why what I did worked. 

My question is that the first (keepalived::vrrp_instance:) is picked up in my puppet/foreman setup without any modifications, other than including keepalived class in a manifest for the node type I have defined as lvs. 

The second (keepalived::lvs_virtual_server:) however I needed the following line in the manifest for the node type.

create_resources(keepalived::lvs::virtual_server, hiera(keepalived::lvs_virtual_server), {})

Now I am assuming that this has something to do with the following in the keepalived module.

 create_resources(keepalived::vrrp::instance, $::keepalived::vrrp_instance)

Which occurs in the config.pp, but just using

 create_resources(keepalived::vrrp::instance, $::keepalived::lvs_virtual_server)

Did not work, hence the hiera call in my code. 

So my questions are why this is the case, and what am I doing wrong/right?

Thanks, 

Hiera below. 

keepalived::vrrp_instance:
  VI_1:
    interface: 'enp2s0'
    state: 'MASTER'
    virtual_router_id: '1'
    priority: '100'
    virtual_ipaddress: '192.168.1.1/24'
    track_interface:
      - 'enp2s0'

keepalived::lvs_virtual_server:
  gftp:
    ip_address: '192.168.1.1'
    port: '2811'
    protocol: 'TCP'
    lb_algo: 'wlc'
    lb_kind: 'DR'
    tcp_check:
      connect_timeout: 3
    delay_loop: '10'
    real_servers:
      - ip_address: '192.168.1.2'
        port: '2811'

### PR DESCRIPTION
**Pull request by 7matd7**
#### Pull Request (PR) description
Adds option to configure LVS real and virtual servers from hiera level

#### This Pull Request (PR) fixes the following issues
Fixes #116 ?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -377,6 +377,39 @@ keepalived::lvs::real_server { 'example2.example.com':
...
 ```
 
+or hiera:
+
+```yaml
+---
+keepalived::lvs_virtual_server:
+  www.example.com:
+    ip_address: '1.2.3.4'
+      port: 80
```
Comment (rubenk) [inline README.md]: indentation is off here

File: README.md [hunk 2]
```diff
@@ -377,6 +377,39 @@ keepalived::lvs::real_server { 'example2.example.com':
...
+
+keepalived::lvs_real_server:
+  example1.example.com:
+    virtual_server: 'www.example.com'
+    ip_address: '1.2.3.8'
+    port: 80
+    options:
+      weight: '1000'
+      TCP_CHECK:
+        connection_timeout: 3
```
Comment (rubenk) [inline README.md]: it is connect_timeout, not connection_timeout

File: README.md [hunk 3]
```diff
@@ -377,6 +377,39 @@ keepalived::lvs::real_server { 'example2.example.com':
...
+  example2.example.com:
+    virtual_server: 'www.example.com'
+    ip_address: '1.2.3.9'
+    port: 80
+    options:
+      weight: '1000'
+      TCP_CHECK:
+        connection_timeout: 3
```
Comment (rubenk) [inline README.md]: same here

**Comment by rubenk** [comment]
Just tested this on CentOS 7, works great! I left a few minor comments.

File: README.md (hunk 1 above)
Comment (7matd7) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (7matd7) [reply README.md]: fixed

**Comment by bastelfreak** [comment]
Thanks for the work!

**Pull request merged**